### PR TITLE
Some misc mappings

### PR DESCRIPTION
**Pull request by ravenbuilder934**
- some of `net.minecraft.stats`
 - `net.minecraft.world.damagesource`
 - `net.minecraft.world.effect`
 - `net.minecraft.world.food`
 - `net.minecraft.world.scores`

### CODE REVIEW

**Review by jriwanek — DISMISSED** [review]

File: data/net/minecraft/stats/ServerStatsCounter.mapping [hunk 1]
```diff
@@ -1,4 +1,19 @@
 CLASS net/minecraft/stats/ServerStatsCounter
+	METHOD <init> (Lnet/minecraft/server/MinecraftServer;Ljava/io/File;)V
+		ARG 1 server
+		ARG 2 file
+	METHOD fromJson (Lcom/google/gson/JsonObject;)Lnet/minecraft/nbt/CompoundTag;
+		ARG 0 tag
```
Comment (Lanse505) [inline data/net/minecraft/stats/ServerStatsCounter.mapping]: ``tag`` -> ``tagObject``

File: data/net/minecraft/stats/ServerStatsCounter.mapping [hunk 2]
```diff
@@ -1,4 +1,19 @@
...
+	METHOD getKey (Lnet/minecraft/stats/Stat;)Lnet/minecraft/resources/ResourceLocation;
+		ARG 0 stat
+	METHOD getStat (Lnet/minecraft/stats/StatType;Ljava/lang/String;)Ljava/util/Optional;
+		ARG 1 map
```
Comment (Lanse505) [inline data/net/minecraft/stats/ServerStatsCounter.mapping]: ``map`` -> ``type``

File: data/net/minecraft/stats/ServerStatsCounter.mapping [hunk 3]
```diff
@@ -1,4 +1,19 @@
...
+		ARG 2 json
```
Comment (Lanse505) [inline data/net/minecraft/stats/ServerStatsCounter.mapping]: ``json`` -> ``id``

File: data/net/minecraft/stats/ServerStatsCounter.mapping [hunk 4]
```diff
@@ -1,4 +1,19 @@
...
+	METHOD lambda$toJson$4 (Lnet/minecraft/stats/StatType;)Lcom/google/gson/JsonObject;
+		ARG 0 json
```
Comment (Lanse505) [inline data/net/minecraft/stats/ServerStatsCounter.mapping]: ``json`` -> ``type``

It's returning the type object there not a json object.

File: data/net/minecraft/stats/Stat.mapping [hunk 5]
```diff
@@ -1,7 +1,13 @@
 CLASS net/minecraft/stats/Stat
+	METHOD <init> (Lnet/minecraft/stats/StatType;Ljava/lang/Object;Lnet/minecraft/stats/StatFormatter;)V
+		ARG 1 type
+		ARG 2 value
+		ARG 3 formatter
 	METHOD buildName (Lnet/minecraft/stats/StatType;Ljava/lang/Object;)Ljava/lang/String;
 		ARG 0 type
 		ARG 1 value
+	METHOD equals (Ljava/lang/Object;)Z
+		ARG 1 object
```
Comment (Lanse505) [inline data/net/minecraft/stats/Stat.mapping]: ``object`` -> ``other``

File: data/net/minecraft/stats/StatsCounter.mapping [hunk 6]
```diff
@@ -4,6 +4,7 @@ CLASS net/minecraft/stats/StatsCounter
 		ARG 1 stat
 	METHOD getValue (Lnet/minecraft/stats/StatType;Ljava/lang/Object;)I
 		ARG 1 type
+		ARG 2 item
```
Comment (Lanse505) [inline data/net/minecraft/stats/StatsCounter.mapping]: ``item`` -> ``type``

File: data/net/minecraft/world/damagesource/BadRespawnPointDamage.mapping [hunk 7]
```diff
@@ -2,3 +2,5 @@ CLASS net/minecraft/world/damagesource/BadRespawnPointDamage
 	METHOD getLocalizedDeathMessage (Lnet/minecraft/world/entity/LivingEntity;)Lnet/minecraft/network/chat/Component;
 		COMMENT Gets the death message that is displayed when the player dies
 		ARG 1 livingEntity
+	METHOD lambda$getLocalizedDeathMessage$0 (Lnet/minecraft/network/chat/Style;)Lnet/minecraft/network/chat/Style;
+		ARG 0 modifyFunc
```
Comment (Lanse505) [inline data/net/minecraft/world/damagesource/BadRespawnPointDamage.mapping]: ``modifyFunc`` -> ``componentStyle``

File: data/net/minecraft/world/damagesource/EntityDamageSource.mapping [hunk 8]
```diff
@@ -1,6 +1,9 @@
 CLASS net/minecraft/world/damagesource/EntityDamageSource
 	FIELD isThorns Z
 		COMMENT Whether this EntityDamageSource is from an entity wearing Thorns-enchanted armor.
+	METHOD <init> (Ljava/lang/String;Lnet/minecraft/world/entity/Entity;)V
+		ARG 1 damageType
```
Comment (Lanse505) [inline data/net/minecraft/world/damagesource/EntityDamageSource.mapping]: ``damageType`` -> ``damageTypeId``

File: data/net/minecraft/world/damagesource/IndirectEntityDamageSource.mapping [hunk 9]
```diff
@@ -1,4 +1,8 @@
 CLASS net/minecraft/world/damagesource/IndirectEntityDamageSource
+	METHOD <init> (Ljava/lang/String;Lnet/minecraft/world/entity/Entity;Lnet/minecraft/world/entity/Entity;)V
+		ARG 1 damageType
```
Comment (Lanse505) [inline data/net/minecraft/world/damagesource/IndirectEntityDamageSource.mapping]: ``damageType`` -> ``damageTypeId``

File: data/net/minecraft/world/effect/AttackDamageMobEffect.mapping [hunk 10]
```diff
@@ -1,4 +1,8 @@
 CLASS net/minecraft/world/effect/AttackDamageMobEffect
+	METHOD <init> (Lnet/minecraft/world/effect/MobEffectCategory;ID)V
+		ARG 1 type
```
Comment (Lanse505) [inline data/net/minecraft/world/effect/AttackDamageMobEffect.mapping]: ``type`` -> ``category``

File: data/net/minecraft/world/effect/MobEffectInstance.mapping [hunk 11]
```diff
@@ -3,8 +3,42 @@ CLASS net/minecraft/world/effect/MobEffectInstance
...
+	METHOD <init> (Lnet/minecraft/world/effect/MobEffect;IIZZZLnet/minecraft/world/effect/MobEffectInstance;)V
+		ARG 1 effect
+		ARG 2 duration
+		ARG 3 amplifier
+		ARG 4 ambient
+		ARG 5 visible
+		ARG 6 showIcon
+		ARG 7 hiddenEffect
+	METHOD <init> (Lnet/minecraft/world/effect/MobEffectInstance;)V
+		ARG 1 mobEffect
```
Comment (Lanse505) [inline data/net/minecraft/world/effect/MobEffectInstance.mapping]: ``mobEffect`` -> ``effectInstance``

File: data/net/minecraft/world/effect/MobEffectInstance.mapping [hunk 12]
```diff
@@ -3,8 +3,42 @@ CLASS net/minecraft/world/effect/MobEffectInstance
...
 	METHOD applyEffect (Lnet/minecraft/world/entity/LivingEntity;)V
 		ARG 1 entity
+	METHOD equals (Ljava/lang/Object;)Z
+		ARG 1 hashCode
```
Comment (Lanse505) [inline data/net/minecraft/world/effect/MobEffectInstance.mapping]: ``hashCode`` -> ``other``

File: data/net/minecraft/world/effect/MobEffectInstance.mapping [hunk 13]
```diff
@@ -20,11 +54,14 @@ CLASS net/minecraft/world/effect/MobEffectInstance
 	METHOD save (Lnet/minecraft/nbt/CompoundTag;)Lnet/minecraft/nbt/CompoundTag;
 		COMMENT Write a custom potion effect to a potion item's NBT data.
 		ARG 1 nbt
+	METHOD setDetailsFrom (Lnet/minecraft/world/effect/MobEffectInstance;)V
+		ARG 1 mobEffect
```
Comment (Lanse505) [inline data/net/minecraft/world/effect/MobEffectInstance.mapping]: ``mobEffect`` -> ``effectInstance``

File: data/net/minecraft/world/effect/MobEffectInstance.mapping [hunk 14]
```diff
@@ -20,11 +54,14 @@ CLASS net/minecraft/world/effect/MobEffectInstance
...
 	METHOD setNoCounter (Z)V
 		COMMENT Toggle the isPotionDurationMax field.
 		ARG 1 maxDuration
 	METHOD tick (Lnet/minecraft/world/entity/LivingEntity;Ljava/lang/Runnable;)Z
 		ARG 1 entity
+		ARG 2 runnable
```
Comment (Lanse505) [inline data/net/minecraft/world/effect/MobEffectInstance.mapping]: ``runnable`` -> ``onExpirationRunnable``

File: data/net/minecraft/world/scores/Objective.mapping [hunk 15]
```diff
@@ -0,0 +1,13 @@
+CLASS net/minecraft/world/scores/Objective
+	METHOD <init> (Lnet/minecraft/world/scores/Scoreboard;Ljava/lang/String;Lnet/minecraft/world/scores/criteria/ObjectiveCriteria;Lnet/minecraft/network/chat/Component;Lnet/minecraft/world/scores/criteria/ObjectiveCriteria$RenderType;)V
+		ARG 1 scoreboard
+		ARG 2 name
+		ARG 3 criteria
+		ARG 4 displayName
+		ARG 5 renderType
+	METHOD lambda$createFormattedDisplayName$0 (Lnet/minecraft/network/chat/Style;)Lnet/minecraft/network/chat/Style;
+		ARG 1 modifyFunc
```
Comment (Lanse505) [inline data/net/minecraft/world/scores/Objective.mapping]: ``modifyFunc`` -> ``style``

**Review by Lanse505 — CHANGES_REQUESTED** [review]

**Review by Lanse505 — APPROVED** [review]

**Review by jriwanek — APPROVED** [review]

**Pull request merged**